src/uniswap.act.md

```diff
@@ -535,6 +535,1073 @@ calls
+performance optimization to keep proof times somewhat reasonable.
+
+#### `mint`: no fee minted
+
+```act
+behaviour mint-noFee-first-diff of UniswapV2Pair
+interface mint(address to)
+
+for all
+
+    Token0 : address UniswapV2Pair
+    Token1 : address UniswapV2Pair
+
+    Balance0 : uint256
+    Balance1 : uint256
+    Reserve0 : uint112
+    Reserve1 : uint112
+
+    PriceLast0 : uint256
+    PriceLast1 : uint256
+    BlockTimestampLast : uint32
+
+    Factory : address UniswapV2Factory
+    FeeTo   : address
+    KLast   : uint256
+
+    DstBal      : uint256
+    Burned      : uint256
+    TotalSupply : uint256
+
+    LockState : uint256
+
+where
+
+    // --- input token amounts ---
+
+    Amount0 := Balance0 - Reserve0
+    Amount1 := Balance1 - Reserve1
+
+    // --- fee liquidity Calculations ---
+
+    FeeOn := FeeTo =/= 0
+
+    RootK := #sqrt(Reserve0 * Reserve1)
+    RootKLast := #sqrt(KLast)
+    FeeLiquidity := (TotalSupply * (RootK - RootKLast)) / ((RootK * 5) + RootKLast)
+
+    // --- LP share minting ---
+
+    MINIMUM_LIQUIDITY := 1000
+
+    // --- time elapsed since last block in which _update was called ---
+
+    TimeElapsed := ((TIME mod pow32) -Word BlockTimestampLast) mod pow32
+
+storage Token0
+
+    balanceOf[ACCT_ID] |-> Balance0
+
+storage Token1
+
+    balanceOf[ACCT_ID] |-> Balance1
+
+storage Factory
+
+    feeTo |-> FeeTo
+
+storage
+
+    token0  |-> Token0
+    token1  |-> Token1
+    factory |-> Factory
+
+    // --- reentrancy guard ---
+
+    lockState |-> LockState => LockState
+
+    // --- cache invariant ---
+
+    kLast |-> KLast => #if FeeOn #then Balance0 * Balance1 #else 0 #fi
+
+    // --- mint tokens ---
+
+    balanceOf[to] |-> DstBal      => DstBal + #sqrt(Amount0 * Amount1) - MINIMUM_LIQUIDITY
+    balanceOf[0]  |-> Burned      => Burned + MINIMUM_LIQUIDITY
+    totalSupply   |-> TotalSupply => TotalSupply + #sqrt(Amount0 * Amount1)
+
+    // --- sync reserves to balances, update cached timestamp ---
+
+    reserve0_reserve1_blockTimestampLast |-> #WordPackUInt112UInt112UInt32(Reserve0, Reserve1, BlockTimestampLast) \
+      => #WordPackUInt112UInt112UInt32(Balance0, Balance1, (TIME mod pow32))
+
+    // --- price accumulator updates ---
+
+    price0CumulativeLast |-> PriceLast0 =>                                          \
+      #if Reserve0 =/= 0 and Reserve1 =/= 0 and TimeElapsed > 0                     \
+        #then chop(((((pow112 * Reserve1) / Reserve0) * TimeElapsed) + PriceLast0)) \
+        #else PriceLast0                                                            \
+      #fi
+
+    price1CumulativeLast |-> PriceLast1 =>                                          \
+      #if Reserve0 =/= 0 and Reserve1 =/= 0 and TimeElapsed > 0                     \
+        #then chop(((((pow112 * Reserve0) / Reserve1) * TimeElapsed) + PriceLast1)) \
+        #else PriceLast1                                                            \
+      #fi
+
+iff in range uint112
+
+    Balance0
```

Comment:
What's the difference with "declaring" this as a `uint112` under `for all`?

Reply:
Semantically I think none, but I think that declaring it a `uint256` and then constraining it further to be a `uint112` better reflects the intention of the source code.